src/transforms/stdnames.jl

```diff
@@ -0,0 +1,61 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    StdNames(:spec)
+
+StdNames transform standardizes the column names.
+"""
+struct StdNames <: Stateless
+  spec::Symbol
+end
+
+StdNames() = StdNames(:upper)
+
+isrevertible(::Type{StdNames}) = true
+
+function apply(transform::StdNames, table)  
+  cols = Tables.columns(table)
+  oldnames = Tables.columnnames(cols)
+  
+  spec = transform.spec
+
+  spec == :camel && (newstringnames = _camel.(string.(oldnames)))
+  spec == :snake && (newstringnames = _snake.(string.(oldnames)))
+  spec == :upper && (newstringnames = _upper.(string.(oldnames)))
+
+  newuniquenames = _unique(newstringnames)
+  newnames = Symbol.(newuniquenames)
+  names = Dict(zip(oldnames, newnames))
```

Comment:
Let's first work on a PR to add a Rename with Dict of String, then we can come back here and avoid all this back and forth between Symbol and String.